### PR TITLE
Implemented Add Points endpoint

### LINKED ISSUE #29: Add points
## Task(s)
- [x] For Discord: check if messages limit reached (5 per day)
- [x] For event: check if Point with eventId already exist, reject if so
- [x] Add a Point to table with appropriate fields
- [x] ~~Increment total points for the User~~

### LINKED ISSUE #35: Check Secret Code for event
## Task(s)
- [x] In the Add Points route, check for the user's input against the event's secret code

### PR DESCRIPTION
**Pull request by cqvu**
## Description

- Implemented Add Points endpoint

Fixes #29 
Fixes #35 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots
N/A

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -147,6 +148,62 @@ def discord_callback():
...
+    amount = data['amount']
+    assignee = data['assignee']
+    description = data['description']
+    event_id = None
+
+    if description == 'Event':
+        event_id = data['event_id']
+        # Check if points claimed for event already
+        if Points.query.filter_by(event_id=event_id, assignee=assignee).first():
+            message = 'Event points already claimed.'
```
Comment (boyuan12) [inline backend/app.py:167]: Check secret code

Comment (cqvu) [reply backend/app.py]: Nvm included it in this PR @boyuan12

**Review by luiszugasti — COMMENTED** [review]
Looks good :rocket:

File: backend/app.py [hunk 2]
```diff
@@ -157,6 +161,92 @@ def discord_callback():
...
+
+    # Add to user's total points
+    user.points_total += amount
+
+    db.session.commit()
+
+    return jsonify({
+        "success": success,
+        "message": message
+    })
```
Comment (PawanKolhe) [inline backend/app.py:247]: Could you send back the amount (amount of points added), discord_id/handle, and timestamp as JSON. Really need it for when the description is an 'Event' and the client needs to know how many points were added.
Maybe something like this:
```json
{
    "success": success,
    "message": message,
    "data": {
        "amount": amount,
        "assignee": assignee,
        "timestamp": ...
    }
}
```

**Review by PawanKolhe — CHANGES_REQUESTED** [review]

File: backend/app.py (hunk 2 above)
Comment (cqvu) [reply backend/app.py]: Added @PawanKolhe

**Review by PawanKolhe — APPROVED** [review]

**Pull request merged**